mackerel-plugin-aws-ses/lib/aws-ses.go

```diff
@@ -40,53 +44,84 @@ var graphdef = map[string]mp.Graphs{
 
 // SESPlugin mackerel plugin for Amazon SES
 type SESPlugin struct {
+	Region          string
 	Endpoint        string
 	AccessKeyID     string
 	SecretAccessKey string
 }
 
 // FetchMetrics interface for mackerel plugin
 func (p SESPlugin) FetchMetrics() (map[string]float64, error) {
-	if p.Endpoint == "" {
-		return nil, errors.New("no endpoint")
+	sess, err := session.NewSession()
+	if err != nil {
+		return nil, err
+	}
+
+	config := aws.NewConfig()
+
+	if p.AccessKeyID != "" && p.SecretAccessKey != "" {
+		config = config.WithCredentials(credentials.NewStaticCredentials(p.AccessKeyID, p.SecretAccessKey, ""))
+	}
+	if p.Region != "" && p.Endpoint != "" {
+		return nil, errors.New("--region and --endpoint are exclusive")
+	}
+	if p.Region != "" {
+		config = config.WithRegion(p.Region)
+	}
+	// for backward compatibility
+	if p.Endpoint != "" {
```

Comment:
It's better to test this parsing behavior, at least valid case (`https://email.us-west-2.amazonaws.com` => `us-west-2`)

Reply:
added at 9449001ab68f626c94996db2932a29941363d757